docs/guides/advanced/libraries.md

```diff
@@ -1,3 +1,50 @@
 # Using libraries
 
-TODO
+JavaScript has a rich and diverse set of libraries available, making it easy to re-use the work of others and more quickly develop an application. You can utilize many of these libraries when building Packs, but there are some important limitations to be aware of.
+
+## Only available in the CLI
+
+At the moment you can only add libraries to your Pack when building with the Packs command line tool (CLI). If you are using the browser-based Pack Studio and want to use a library you'll have to migrate to the CLI workflow. See the [CLI getting started guide][get_started_cli] for information on how to install and setup the CLI.
+
+## Use NPM
+
+The easiest way to install libraries is via the Node package manager (NPM). While originally created just for the Node.js environment, it's now used to distribute all sorts of JavaScript libraries. If you completed the [CLI getting started guide][get_started_cli] then you should already have NPM installed.
+
+To add a library to your Pack simply use `npm install` to install it. For example, to install the popular JavaScript library [Lodash][lodash] you would run:
+
+```sh
+npm install lodash
+```
+
+You can then import this library into your code using either Node's `require` or the more modern `import` syntax:
+
+```ts
+const _ = require('lodash');
+// ... or ...
+import _ from 'lodash';
+```
+
+!!! info
+    How to import the library may vary; consult the library's documentation for the correct syntax.
+
+[get_started_cli]: ../../get-started/cli.md
+[lodash]: https://lodash.com/
+
+## Library compatibility
+
+Some JavaScript libraries are written assuming the code is being run in either a web browser or on a Node.js server, and the Packs execution environment isn't exactly either. The execution environment provides access to the [standard built-in JavaScript objects][mdn_standard], but doesn't support [Web APIs][mdn_web] or [Node.js APIs][node_apis]. These include but are not limited to:
+
+- Browser: `window`, `document`, `XMLHttpRequest`, `fetch()`
+- Node.js: `fs`, `http`
+
+Unfortunately there isn't an easy way to determine beforehand if a given library will fail due to an unavailable API. At the moment the best approach is to try the library and see if it runs successfully in the Packs environment.
```

Comment:
You may want to expand on this (could be in a followup), the maker can e.g. use `coda execute` which will as of this week by default run your code in our VM with all the restrictions in place, and you should be able to see at that point what works and what doesn't.

Another potential followup would be to talk about how setTimeout is disabled by default, e.g. what gets explained here: https://github.com/coda/packs-sdk/blob/main/testing/injections/timers_disabled_shim.js#L5

Reply:
I've added a line about `coda execute`, and it's a good callout. I started investigating the timers shims, but found some issues, so I'll address in a follow-up PR.